From ff981c63c88936d52f76e8604f9a06e38191741e Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Sat, 10 Jan 2015 11:36:51 +0100
Subject: [PATCH 1/5] rename phonon meta include file in the source and instead
 rename on-install

having phonon and Phonon in the same source dir screws with
case-insensitive operating systems.

note: this still is a problem if one installs as on the target it would
still be the same setup with both cased files in the same directory.
searching for the actual include lineup that debian was trying to solve
appears as though it actualy never existed, so I am actually not sure
where one should put the header to avoid this. perhaps it'd be best to
simply not offer the compat option on !linux. it does certainly make a
lot less sense on other systems anyway.

CCMAIL: kde-mac@kde.org
---
 includes/CMakeLists.txt          | 18 +++++++++++++++---
 includes/old/phonon              | 35 -----------------------------------
 includes/old/phonon_meta_include | 35 +++++++++++++++++++++++++++++++++++
 3 files changed, 50 insertions(+), 38 deletions(-)
 delete mode 100644 includes/old/phonon
 create mode 100644 includes/old/phonon_meta_include

diff --git a/includes/CMakeLists.txt b/includes/CMakeLists.txt
index 857fd4e..c27376a 100644
--- a/includes/CMakeLists.txt
+++ b/includes/CMakeLists.txt
@@ -91,10 +91,22 @@ set(COMPAT_HEADERS
     old/Phonon/VolumeFaderInterface
     old/Phonon/VolumeSlider
 )
-install(FILES ${COMPAT_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/KDE/Phonon COMPONENT Devel)
+
+install(FILES ${COMPAT_HEADERS}
+    DESTINATION ${INCLUDE_INSTALL_DIR}/KDE/Phonon
+    COMPONENT Devel)
+# FIXME: phonon/Phonon (dir) and phonon/phonon (file) conflict on case-insensitive
+#        systems if one enables this
 if (PHONON_INSTALL_QT_COMPAT_HEADERS)
-    install(FILES ${COMPAT_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/phonon/Phonon COMPONENT Devel) # <Phonon/Foo>
-    install(FILES old/phonon DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel) # <phonon/phonon> meta-include
+    # <Phonon/Foo>
+    install(FILES ${COMPAT_HEADERS}
+        DESTINATION ${INCLUDE_INSTALL_DIR}/phonon/Phonon
+        COMPONENT Devel)
+    # <phonon/phonon> meta-include
+    install(FILES old/phonon_meta_include
+        DESTINATION ${INCLUDE_INSTALL_DIR}/phonon
+        COMPONENT Devel
+        RENAME phonon)
 endif()
 
 if (PHONON_BUILD_EXPERIMENTAL)
diff --git a/includes/old/phonon b/includes/old/phonon
deleted file mode 100644
index 82522bf..0000000
--- a/includes/old/phonon
+++ /dev/null
@@ -1,35 +0,0 @@
-// This is a compatibility include for QtPhonon. It should not be used
-// in new software!
-
-#include "abstractaudiooutput.h"
-#include "abstractmediastream.h"
-#include "abstractvideooutput.h"
-#include "addoninterface.h"
-#include "audiooutput.h"
-#include "audiooutputinterface.h"
-#include "backendcapabilities.h"
-#include "backendinterface.h"
-#include "effect.h"
-#include "effectinterface.h"
-#include "effectparameter.h"
-#include "effectwidget.h"
-#include "globaldescriptioncontainer.h"
-#include "mediacontroller.h"
-#include "medianode.h"
-#include "mediaobject.h"
-#include "mediaobjectinterface.h"
-#include "mediasource.h"
-#include "objectdescription.h"
-#include "objectdescriptionmodel.h"
-#include "path.h"
-#include "phonondefs.h"
-#include "phononnamespace.h"
-#include "platformplugin.h"
-#include "seekslider.h"
-#include "streaminterface.h"
-#include "videoplayer.h"
-#include "videowidget.h"
-#include "videowidgetinterface.h"
-#include "volumefadereffect.h"
-#include "volumefaderinterface.h"
-#include "volumeslider.h"
diff --git a/includes/old/phonon_meta_include b/includes/old/phonon_meta_include
new file mode 100644
index 0000000..82522bf
--- /dev/null
+++ b/includes/old/phonon_meta_include
@@ -0,0 +1,35 @@
+// This is a compatibility include for QtPhonon. It should not be used
+// in new software!
+
+#include "abstractaudiooutput.h"
+#include "abstractmediastream.h"
+#include "abstractvideooutput.h"
+#include "addoninterface.h"
+#include "audiooutput.h"
+#include "audiooutputinterface.h"
+#include "backendcapabilities.h"
+#include "backendinterface.h"
+#include "effect.h"
+#include "effectinterface.h"
+#include "effectparameter.h"
+#include "effectwidget.h"
+#include "globaldescriptioncontainer.h"
+#include "mediacontroller.h"
+#include "medianode.h"
+#include "mediaobject.h"
+#include "mediaobjectinterface.h"
+#include "mediasource.h"
+#include "objectdescription.h"
+#include "objectdescriptionmodel.h"
+#include "path.h"
+#include "phonondefs.h"
+#include "phononnamespace.h"
+#include "platformplugin.h"
+#include "seekslider.h"
+#include "streaminterface.h"
+#include "videoplayer.h"
+#include "videowidget.h"
+#include "videowidgetinterface.h"
+#include "volumefadereffect.h"
+#include "volumefaderinterface.h"
+#include "volumeslider.h"
-- 
1.9.3

